vowpalwabbit/feature_group.h

```diff
@@ -289,10 +365,14 @@ struct features
   using const_iterator = features_iterator<const feature_value, const feature_index>;
   using audit_iterator = audit_features_iterator<feature_value, feature_index, audit_strings>;
   using const_audit_iterator = audit_features_iterator<const feature_value, const feature_index, const audit_strings>;
+  using extent_iterator = ns_extent_iterator<features, audit_iterator, std::vector<VW::namespace_extent>::iterator>;
+  using const_extent_iterator =
+      ns_extent_iterator<const features, const_audit_iterator, std::vector<VW::namespace_extent>::const_iterator>;
 
-  v_array<feature_value> values;               // Always needed.
-  v_array<feature_index> indicies;             // Optional for sparse data.
-  std::vector<audit_strings> space_names;      // Optional for audit mode.
+  v_array<feature_value> values;           // Always needed.
+  v_array<feature_index> indicies;         // Optional for sparse data.
```

Comment:
nit: Is this v_array still optional as per the comment?

Reply:
Yes, but in practice it is almost always used